Mathlib/Analysis/SumOverResidueClass.lean

```diff
@@ -0,0 +1,93 @@
+/-
+Copyright (c) 2024 Michael Stoll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll
+-/
+import Mathlib.Analysis.NormedSpace.FiniteDimension
+
+/-!
+# Sums over residue classes
+
+We consider infinite sums over functions `f` on `ℕ`, restricted to a residue class mod `m`.
+
+The main result is `summable_indicator_mod_iff`, which states that when `f : ℕ → ℝ` is
+decreasing and takes nonnegative values, then the sum over `f` restricted to any residue class
+mod `m ≠ 0` converges if and only if the sum over all of `ℕ` converges.
+
+(Note that we want to use `Summable.of_nonneg_of_le`, which insists on `ℝ` as the target.)
```

Comment:
I'd remove this comment, as I don't see what other target would really make sense (nonnegative reals, maybe, but then it's easy to deduce from the real result).

Reply:
OK. (I was thinking that maybe there is a more general version possible, but maybe the reals are essentially the only relevant possibility here).